add pure-python (async) HTTP request how to

emirkmo opened the pull request:
This PR adds a How-to on making HTTP requests from within pyscript, using pyfetch in Pyodide, which is a wrapper around the `fetch` API. (Read the PR how to if you want to learn more!) The inability to use requests/httpx is one of the first and most common questions by newcomers, hence the need.

This How-to was suggested by https://github.com/pyscript/pyscript/pull/151#issuecomment-1243714175 based on the related community example by me in #151. 

Edit: I verified the output on Sphinx autobuilt docs and on GitHub. It looks good to me!

madhur-tandon commented on docs/howtos/http-requests.md:
```diff
@@ -0,0 +1,170 @@
+
+The `fetch` API is a modern way to make HTTP requests. It is available in all modern browsers, and in Pyodide.
+
+Although there are two ways to use `fetch`, 1) using javacript from pyscript, and 2) using Pyodide's python wrapper,
```
`javacript` --> `javascript`

emirkmo replied:
Good call, will do the latter.

emirkmo replied:
Thanks for the catch and suggestion, I've made the changes. Looks better.

emirkmo commented:
@madhur-tandon I implemented your review, and made some changes to clarity by expounding upon the use of `pyfetch` and HTTP requests. I also added an API Reference section at the end, for those who don't want to read the whole How-to.

@marimeireles I have finalized the PR with the How-to, please let me know if this is what you had in mind.

marimeireles commented on docs/howtos/http-requests.md:
```diff
@@ -0,0 +1,210 @@
+for more information.
+
+# Example
+We will make async HTTP requests to [JSONPlaceholder](https://jsonplaceholder.typicode.com/) fake API using `pyfetch`.
```
We will make async HTTP requests to [JSONPlaceholder](https://jsonplaceholder.typicode.com/)'s fake API using `pyfetch`.

emirkmo replied:
Clarified what I meant

marimeireles commented on docs/howtos/http-requests.md:
```diff
@@ -0,0 +1,210 @@
+The basic idea is that the `PyScript` will import and call this function, then await the response. Therefore,
+the script containing this function must be importable by `PyScript`.
+
+For this example, we will name the file `request.py` and place it in the same directory as the file
+containing the html code below.
```
code bellow -> code above
(?)

emirkmo replied:
Fixed intent

marimeireles reviewed (commented):
Woh! I love it :D
Thank you so much for the contribution!
Some small nitpicks from my side and another annoying one, we should probably change python -> Python.
Let me know if it's too annoying and I can fix that.
Thank you so much for the great work! Pretty sure it will help lots of folks ❤️

emirkmo commented:
> Woh! I love it :D Thank you so much for the contribution! Some small nitpicks from my side and another annoying one, we should probably change python -> Python. Let me know if it's too annoying and I can fix that. Thank you so much for the great work! Pretty sure it will help lots of folks heart

Ok, I changed python -> Python or `Python` depending on whether it was in the Tutorial or not. Not sure this is what you asked exactly, but feel free to make changes if you find anything else.

Cheers

emirkmo commented:
@marimeireles anything left blocking approval?  :)

marimeireles commented:
@emirkmo just me forgetting to click the button.
Thanks for the ping!
